atak-civ/plugin-examples/plugintemplate/app/src/main/java/com/atakmap/android/plugintemplate/CreateTimerDropDown.java

```diff
@@ -0,0 +1,176 @@
+
+import android.content.Context;
+import android.content.Intent;
+import android.view.View;
+import android.widget.Button;
+import android.widget.CheckBox;
+import android.widget.EditText;
+
+import com.atak.plugins.impl.PluginLayoutInflater;
+import com.atakmap.android.dropdown.DropDownReceiver;
+import com.atakmap.android.ipc.AtakBroadcast;
+import com.atakmap.android.maps.MapView;
+import com.atakmap.android.plugintemplate.plugin.R;
+import com.atakmap.coremap.log.Log;
+import com.atakmap.android.dropdown.DropDown.OnStateListener;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+
+public class CreateTimerDropDown extends DropDownReceiver implements OnStateListener {
+
+    public static final String TAG = PluginTemplateDropDownReceiver.class
+            .getSimpleName();
+
+    // set up the string so that when called the dropdown reciever knows what to open
+    public static final String SHOW_CREATE = "com.atakmap.android.plugintemplate.CreateTimerDropDown";
+    private final View templateView;
+    private final Context pluginContext;
+    private Timer timer;
+    private EditText name;
+    private EditText durationHours;
+    private EditText durationMinutes;
+    private EditText durationSeconds;
+    private CheckBox preset;
+    private Button changeSoundButton;
+    public CreateTimerDropDown(final MapView mapView,
+                                final Context context) {
+        super(mapView);
+        this.pluginContext = context;
+        timer = new Timer();
+
+
+        // Remember to use the PluginLayoutInflator if you are actually inflating a custom view
+        // In this case, using it is not necessary - but I am putting it here to remind
+        // developers to look at this Inflator
+        templateView = PluginLayoutInflater.inflate(context, R.layout.create_timer_layout, null);
+//        Button changeRepeatButton = (Button)templateView.findViewById(R.id.changeRepeatButton);
+//        changeRepeatButton.setOnClickListener(new View.OnClickListener() {
+//            @Override
+//            public void onClick(View v) {
+//                Intent i = new Intent();
+//                i.setAction();
+//                AtakBroadcast.getInstance().sendBroadcast(i);
+//            }
+//        });
+        this.name = templateView.findViewById(R.id.timerName);
+        this.durationHours  = templateView.findViewById(R.id.durationHours);
+        this.durationMinutes  = templateView.findViewById(R.id.durationMinutes);
+        this.durationSeconds  = templateView.findViewById(R.id.durationSeconds);
+        this.preset = templateView.findViewById(R.id.checkBox);
+        this.changeSoundButton = (Button)templateView.findViewById(R.id.changeSoundButton);
+
+        changeSoundButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                Intent i = new Intent();
+                i.putExtra("PAGE_TO_RETURN_TO", "CreateTimerDropDown");
```

Comment:
I think you should add "i.putExtra("DEFAULT_SELECTED_SOUND", timer.getSound());" over here so that the select sound screen has the correct value selected when you navigate to it

Reply:
done